### ISSUE TITLE
Document that named query parameters are positional parameters

### ISSUE THREAD

**Issue by ryanheise** [vitusortner/floor#402]
The following doesn't do what people might expect given a prior understanding of how named parameters get bound:

```dart
@Query("update foo set prop = :newValue where prop = :oldValue")
Future<void> updateFooProp(int oldValue, int newValue);
```

It will be converted to `update foo set prop = ? where prop = ?` with values in parameter order: `[oldValue, newValue]`.

I'm sure you're on top of this for a future release, but in the short term it could be a life saver to document this behaviour.

Maybe related: #142 , #321 .

**Comment by vitusortner** [comment]
Closing this issue as it's a duplicate of #224.

**Comment by ryanheise** [comment]
Understood, although my issue was more about documenting the current behaviour rather than implementing new behaviour. Documenting the current behaviour could help save hours of development time while implementing new behaviour may be more of a long term thing.

**Comment by vitusortner** [comment]
We'll fix this behavior with #509, which resolves the need for extra documentation.
